api/v1alpha1/tinkerbell_type.go

```diff
@@ -0,0 +1,125 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=tinkerbell,scope=Namespaced,categories=tinkerbell,singular=tinkerbell,shortName=tb
+// +kubebuilder:storageversion
+
+// Tinkerbell represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
+// Tinkerbell operator watches for different resources such as deployment, services, serviceAccounts, etc. One of those
+// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// the operator will delete all tinkerbell resources.
+type Tinkerbell struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec describes the desired tinkerbell stack state.
+	Spec TinkerbellSpec `json:"spec,omitempty"`
+}
+
+// TinkerbellSpec specifies details of an Tinkerbell setup.
+type TinkerbellSpec struct {
+	// TinkerbellVersion is the tinkerbell crd version.
+	TinkerbellVersion string `json:"tinkerbellVersion"`
+
+	// ClusterDNS is the ip address of the cluster dns resolver.
+	ClusterDNS string `json:"clusterDNS"`
+
+	// OverwriteRegistry is the registry to use for all images. If this field is set, all tink service deployment images
+	// will be prefixed with this value. For example if the value here was set to docker.io, then boots image will be
+	// docker.io/tinkerbell/boots.
+	OverwriteRegistry string `json:"overwriteRegistry"`
+
+	// DockerPullConfig the secret name containing the docker auth config which should exist in the same namespace where
+	// the operator is deployed(typically tinkerbell)
+	DockerPullComfig string `json:"dockerPullComfig"`
```

Comment:
Is this meant to be semantically equivalent to [image pull secrets](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#containers)? I think it would benefit us to use the same as the `pod` spec: `imagePullSecrets` and making it a slice.

Reply:
Yes, and yeah I agree, having the same pull secret semantics that is from k8s native CRs(e.g: Deployments) would make total sense